### ISSUE TITLE
Crash on scan grid order by class

### ISSUE THREAD

**Issue by Hurricane235** [EDDiscovery/EDDiscovery#3497]
### Description
Added Scan Grid tab and sorted by class.

### Exception Details:
>EDDiscovery v17.0.4.0 There was an unhandled UI exception.
>```
>==== BEGIN ====
>System.ArgumentException: Object must be of type String.
>   at System.String.CompareTo(Object value)
>   at System.Collections.Comparer.Compare(Object a, Object b)
>   at System.Windows.Forms.DataGridViewRowCollection.RowComparer.CompareObjects(Object value1, Object value2, Int32 rowIndex1, Int32 rowIndex2)
>   at System.Windows.Forms.DataGridViewRowCollection.RowArrayList.CustomQuickSort(Int32 left, Int32 right)
>   at System.Windows.Forms.DataGridViewRowCollection.Sort(IComparer customComparer, Boolean ascending)
>   at System.Windows.Forms.DataGridView.SortInternal(IComparer comparer, DataGridViewColumn dataGridViewColumn, ListSortDirection direction)
>   at System.Windows.Forms.DataGridView.OnColumnHeaderMouseClick(DataGridViewCellMouseEventArgs e)
>   at System.Windows.Forms.DataGridView.OnMouseClick(MouseEventArgs e)
>   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
>   at System.Windows.Forms.Control.WndProc(Message& m)
>   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
>===== END =====
>```

**Comment by robbyxp1** [comment]
Not reproducible at the moment. 
There is no specialised sort comparer for this grid
All columns are text boxes.

Any information in what was in the grid when this happened?

**Comment by Hurricane235** [comment]
> Not reproducible at the moment. There is no specialised sort comparer for this grid All columns are text boxes.
> 
> Any information in what was in the grid when this happened?

I've done a bit more testing, and I'm not sure if this is normal behaviour or not, but one on the bodies in the grid has no data, Class and Distance are populated with ?. But if I select to pull the data from EDSM so the fields contain actual data, I can sort the columns without crashing.

I'm still relatively new to the game so does this just mean I've not scanned it yet? Or has my local data bugged somewhere?

**Comment by robbyxp1** [comment]
The ? means it knows about the body somehow (ApproachBody etc), but you have not scanned it.  Scanning it would reveal details.

**Comment by Eahlstan** [comment]
I can replicate this, so nothing needed from your side.
Thanks for the report.

**Comment by Eahlstan** [comment]
```
System.ArgumentException
  HResult=0x80070057
  Message=Das Objekt muss den Typ "String" haben.
  Source=mscorlib
  StackTrace:
   at System.String.CompareTo(Object value)
   at System.Windows.Forms.DataGridViewRowCollection.RowComparer.CompareObjects(Object value1, Object value2, Int32 rowIndex1, Int32 rowIndex2)
   at System.Windows.Forms.DataGridViewRowCollection.RowArrayList.Pivot(Int32 left, Int32 center, Int32 right)
   at System.Windows.Forms.DataGridViewRowCollection.RowArrayList.CustomQuickSort(Int32 left, Int32 right)
   at System.Windows.Forms.DataGridViewRowCollection.Sort(IComparer customComparer, Boolean ascending)
   at System.Windows.Forms.DataGridView.SortInternal(IComparer comparer, DataGridViewColumn dataGridViewColumn, ListSortDirection direction)
   at System.Windows.Forms.DataGridView.OnColumnHeaderMouseClick(DataGridViewCellMouseEventArgs e)
   at System.Windows.Forms.DataGridView.OnMouseClick(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG& msg)
   at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   at EDDiscovery.Program.Main() in F:\Git\EDDiscovery\Original\EDDiscovery\EDDiscovery\Program.cs:line 36
```

Happens if you try to sort a column that has the mentioned `?` instead of real data.

**Comment by robbyxp1** [comment]
I can too

**Comment by robbyxp1** [comment]
Thank you, will be fixed in 17.1 out shortly
